### ISSUE TITLE
Boost

### ISSUE THREAD

**Issue by kaan123123** [NadirRoGue/RenderEngine#6]
I am integrating boost library int this project now a days and will finish send u bro, actual your code permutation is good. I wonder 👍  How many days did you cobe the project?? Are u freelancer?

**Comment by NadirRoGue** [comment]
I dont remember how much time I employ doing it haha, it was an assigment for a class in my master's degree. There are many things to be improved on it (memory management, both in CPU and GPU, finish temporal reprojection implementation, replace current basic shading model with PBR, improve some parts of the engine architecture, which either can be done better or are simply hardcoded, etc.)

**Comment by kaan123123** [comment]
yeah yeah bro. very long :)  but I have problem because my computer very slow unlimit terrain, 
I wanna to limit terrain. I dont find unlimit code scene or terrain.
If you have a time, may you look project?

**Comment by NadirRoGue** [comment]
The terrain is rendered as a set of tiles. At each frame, only the tiles in front of the camara are rendered. Changing it to draw a static or limited world wont improve performance. If you want to relieve the computational load, I would start by:
* Reduce the complexity of the tree meshes. This can be modified in `src/terraincomponents/TreeComponent.cpp` (function `initTrees()`). You can start by modifying the max branch depth and max branch per split.
* Reduce the number of trees to spawn per tile. This setting is hardcoded in the same file `src/terraincomponents/TreeComponent.cpp`, in the function `initialize()`. The default value is 12, lower it to reduce the load
* Reduce the number of flowers to spawn per tile. As well as the previous setting, is hardcoded, in the file `src/terraincomponents/FlowerComponent`, function `initialize()`. The default value is 36.
* Reduce the terrain render radius. Currently, on each iteration, the engine is rendering some 250 terrain tiles. You can improve the performance by reducing the render radius. This setting is modified in `src/WorldConfig.cpp`, modifying the variable `worldRenderRadius` (line 14)
